MapleServer2/Packets/ResponseCubePacket.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using MaplePacketLib2.Tools;
+using MapleServer2.Constants;
+using MapleServer2.Types;
+using MapleServer2.Tools;
+using MapleServer2.Servers.Game;
+using MapleServer2.Data.Static;
+using Maple2Storage.Types.Metadata;
+
+namespace MapleServer2.Packets
+{
+    public static class ResponseCubePacket
+    {
+        public static Packet Pickup(GameSession session, byte[] coords)
+        {
+            PacketWriter pWriter = PacketWriter.Of(SendOp.RESPONSE_CUBE);
+
+            // Convert to signed byte array
+            sbyte[] sCoords = Array.ConvertAll(coords, b => unchecked((sbyte) b));
```

Comment:
This should be done in the handler, and instead just pass `sCoords` in as a parameter

Reply:
The packet sends a `byte[]` not an `sbyte[]` so if I convert in handler I'd need to then unconvert in packet.

Reply:
You're only using `sCoords` here to set a new weapon id if it finds one, you can just send the packet method the finished item id, along with the `byte[]` that you need to write